File: Rohit Singhal/Arrays and Strings/StringtoInteger(atoi).java

class Solution {
    public int myAtoi(String s) {
         if(s.length()==0)
         {
             return 0;
         }
        int index=0;
        int sign=1;
        int ans=0;
        while(index<s.length() && s.charAt(index)==' ')
        {
            index++;
        }
        if(index<s.length() &&  (s.charAt(index)=='+' || s.charAt(index)=='-')) 
        {
             if(s.charAt(index)=='-')
           {
            sign=-1;
            index++;
           }
            else
            {
            sign=1;
            index++;
            }
            
        }
       
        while(index<s.length())
        {
            int digit=s.charAt(index)-'0';
            if(digit<0 || digit>9)
                break;
            
            if(Integer.MAX_VALUE/10 < ans || (Integer.MAX_VALUE/10==ans && Integer.MAX_VALUE%10<digit))
            {
                if(sign==1)
                    return Integer.MAX_VALUE;
                else
                    return Integer.MIN_VALUE;
            }
            
            ans=10*ans+digit;
            index++;
        }
        return ans*sign;
        
        
    }
}

File: Hashwanth/Arrays & Strings/String to Integer (atoi)/String to Integer (atoi).cpp
class Solution {
public:
    int myAtoi(string s) {
        if(s.length()==0)
            return 0;
        int i;
        long num=0;
        bool sign=true;
        for(i=0;i<s.length();i++)
        {
            if(s[i]!=' ')
                break;
        }
        if(i==s.length())
            return 0;
        else if(s[i]=='-')
        {
            sign=false;
            i++;
        }
        else if(s[i]=='+')
        {
           
            i++;
        }
        while(i<s.length() && isdigit(s[i]))
        { num=num*10+(s[i]-'0');
            if(num>INT_MAX)
                return sign?INT_MAX:INT_MIN;
           
            i++;
        }
        return sign?num:-1*num;
            
    }
};
